rootfs/registry/dockerclient.py

```diff
@@ -45,26 +45,9 @@ def publish_release(self, source, config, target, deis_registry):
         image = "{}:{}".format(repo, src_tag)
         self.tag(image, src_name, tag=src_tag)
 
-        # build a Docker image that adds a "last-mile" layer of environment
-        config.update({'DEIS_APP': name, 'DEIS_RELEASE': tag})
-        self.build(source, config, name, tag)
-
         # push the image to deis-registry
         self.push("{}/{}".format(self.registry, name), tag)
 
-    def build(self, source, config, repo, tag):
-        """Add a "last-mile" layer of environment config to a Docker image for deis-registry."""
-        check_blacklist(repo)
-        env = ' '.join("{}='{}'".format(
-            k, v.encode('unicode-escape').replace("'", "\\'")) for k, v in config.viewitems())
-        dockerfile = "FROM {}\nENV {}".format(source, env)
-        f = io.BytesIO(dockerfile.encode('utf-8'))
-        target_repo = "{}/{}:{}".format(self.registry, repo, tag)
```

Comment:
By removing this line, we are no longer tagging an image called `$REGISTRY_URL/appname:v2`, which will break the push operation on line 49. From what I see, line 46 doesn't really do much other than re-tag the incoming image from `quay.io/mycustomimage:latest` to just `mycustomimage:latest`, stripping the external registry URL. In order to fix this, I think we need to change line 46 to something like:

```
self.tag(image, "{}/{}".format(self.registry, name), tag=tag)
```

That way the image should be properly bagged and tagged for shipping to the registry. I'll verify my thoughts manually tomorrow, but if you have more insight on how this works then I'm prepared to be wrong :)